sktime/performance_metrics/detection/_randindex.py

```diff
@@ -0,0 +1,230 @@
+import pandas as pd
+
+from sktime.performance_metrics.detection._base import BaseDetectionMetric
+
+
+class RandIndex(BaseDetectionMetric):
+    """Rand Index metric for comparing event detection results.
+
+    By default, if X is provided, this metric computes distances/lengths in loc-based
+    units by looking up the corresponding labels in X.index. Otherwise, or if X is None,
+    it uses the original iloc-based calculations.
+    """
+
+    _tags = {
+        "object_type": ["metric_detection", "metric"],
+        "scitype:y": "segments",
+        "requires_X": False,
+        "requires_y_true": True,
+        "lower_is_better": False,  # Higher Rand Index is better
+    }
+
+    def __init__(self, use_loc=True):
+        """
+        Parameters
+        ----------
+        use_loc : bool, optional (default=True)
+            If True (and X is provided), segment lengths/overlaps are computed as the
+            difference of X.index[end_iloc] - X.index[start_iloc]. If False, or X=None,
+            uses iloc-based distances (end_iloc - start_iloc) as before.
+        """  # noqa: D205
+        self.use_loc = use_loc
+        super().__init__()
+
+    def _evaluate(self, y_true, y_pred, X=None):
+        """Calculate Rand Index between true and predicted segments.
+
+        Parameters
+        ----------
+        y_true : pd.DataFrame
+            Ground truth segments with 'start', 'end', or an 'ilocs' column (interval or int),
+            plus an optional 'label' column.
+        y_pred : pd.DataFrame
+            Predicted segments with 'start', 'end', or an 'ilocs' column (interval or int),
+            plus an optional 'label' column.
+        X : pd.DataFrame, optional (default=None)
+            If provided (and use_loc=True), loc-based distances are used.
+
+        Returns
+        -------
+        float
+            Rand Index score between 0.0 and 1.0.
+        """  # noqa: E501
+        # 1) Validate and extract segments (still iloc-based).
+        y_true_segments = self._extract_segments(y_true, "y_true")
+        y_pred_segments = self._extract_segments(y_pred, "y_pred")
+
+        # 2) Assign unique cluster IDs to each segment.
+        y_true_segments = self._assign_unique_ids(y_true_segments, prefix="true")
```

Comment:
sometimes the segments already come with labels, in this case both dataframes will have a column `label`.

In this case, can we not just use that column?